src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -0,0 +1,322 @@
+    ) {
+      const focusedElement = document.activeElement as HTMLElement
+
+      if (!this._root.current.contains(focusedElement)) {
+        this.focus()
+        ev.preventDefault()
+        ev.stopPropagation()
+      }
+    }
+  }
+
+  private _forceClickInTrap = (ev: MouseEvent): void => {
+    if (
+      FocusTrapZone._focusStack.length &&
+      this === FocusTrapZone._focusStack[FocusTrapZone._focusStack.length - 1]
+    ) {
+      const clickedElement = ev.target as HTMLElement
+
+      if (clickedElement && !this._root.current.contains(clickedElement)) {
+        this.focus()
+        ev.preventDefault()
+        ev.stopPropagation()
+      }
+    }
+  }
+
+  private _subscribeToEvents = () => {
+    const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
+    if (forceFocusInsideTrap) {
+      eventStack.sub('focus', this._forceFocusInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+
+    if (!isClickableOutsideFocusTrap) {
+      eventStack.sub('click', this._forceClickInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+  }
+
+  private _unsubscribeFromEvents = () => {
+    const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
+    if (forceFocusInsideTrap) {
+      eventStack.unsub('focus', this._forceFocusInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+
+    if (!isClickableOutsideFocusTrap) {
+      eventStack.unsub('click', this._forceClickInTrap, {
+        target: this.windowElement,
+        useCapture: true,
+      })
+    }
+  }
+
+  private _getPreviouslyFocusedElementOutsideTrapZone = () => {
+    const { elementToFocusOnDismiss } = this.props
+    let previouslyFocusedElement = this._previouslyFocusedElementOutsideTrapZone
+
+    if (elementToFocusOnDismiss && previouslyFocusedElement !== elementToFocusOnDismiss) {
+      previouslyFocusedElement = elementToFocusOnDismiss
+    } else if (!previouslyFocusedElement) {
+      previouslyFocusedElement = document.activeElement as HTMLElement
+    }
+
+    return previouslyFocusedElement
+  }
+
+  private _hideContentFromAccessibilityTree = () => {
+    const elements = (document.body && document.body.children) || []
+
+    if (elements.length && !document.body.contains(this._root.current)) {
+      // In case popup render options will change
+      throw new Error(
+        'Body does not contain trap zone element as expected. If it is done intentionally, please, make sure to update FocusTrapZone.',
+      )
+    }
+
+    // loop through all body's children, except the last one - which is the popup
+    for (let index = 0; index < elements.length - 1; index++) {
+      const element = elements[index]
+
+      if (
+        element.nodeName !== 'SCRIPT' &&
+        element.nodeName !== 'STYLE' &&
+        element.getAttribute('aria-hidden') !== 'true'
+      ) {
+        element.setAttribute('aria-hidden', 'true')
```

Comment:
don't we want to introduce constant for `aria-hidden` as well? What is the criteria that drives this decision boundary?

Reply:
Usually, we move things to constants when we want to have 1 point for change, in case it will change in the future. 'aria-hidden' will always stay the same and I don't see any reason to create a const instead of a compile-time static string

Reply:
yes, the question then is `which attributes we are deciding to put to constants?` What is the semantics behind them - because even for `aria-hidden`, _in general case_, it is possible to argue that another day we might decide that another attribute should be put to all these places, so we would like to have single point of change. Just to clarify this moment